dbms/src/Interpreters/DDLWorker.cpp

```diff
@@ -684,6 +684,12 @@ void DDLWorker::checkShardConfig(const String & table, const DDLTask & task, Sto
     const auto & shard_info = task.cluster->getShardsInfo().at(task.host_shard_num);
     bool config_is_replicated_shard = shard_info.hasInternalReplication();
 
+    if (storage->isDistributed())
```

Comment:
Maybe we can just check with `dynamic_cast` https://github.com/clickhouse/ClickHouse/blob/master/dbms/src/Interpreters/DDLWorker.cpp#L710?